portal/src/graphql/adminapi/UserDetailsAccountSecurity.tsx

```diff
@@ -63,58 +96,278 @@ function getLocaleKeyWithAuthenticatorType(
   }
 }
 
-function getDescriptionFromAuthenticator(
+function constructPasswordAuthenticatorData(
   authenticator: Authenticator,
   locale: string
-) {
-  const showLastUpdated = authenticator.type === "PASSWORD";
-  if (showLastUpdated) {
-    const lastUpdatedText =
-      formatDatetime(locale, authenticator.updatedAt) ?? "";
-    return {
-      description: (
-        <FormattedMessage
-          id="UserDetails.account-security.last-updated"
-          values={{ datetime: lastUpdatedText }}
-        />
-      ),
-    };
+): PasswordAuthenticatorData {
+  const lastUpdated = formatDatetime(locale, authenticator.updatedAt) ?? "";
+
+  return {
+    id: authenticator.id,
+    kind: authenticator.kind,
+    lastUpdated,
+  };
+}
+
+function getTotpDisplayName(
+  totpAuthenticatorClaims: Record<string, unknown>
+): string {
+  for (const [key, claim] of Object.entries(totpAuthenticatorClaims)) {
+    // expected form: https://authgear.com/claims/totp/display_name
+    const lastSegment = key.split("/").pop();
```

Comment:
You just need to perform a simple string comparison here. The value `https://authgear.com/claims/totp/display_name` is static.